services/management/dbaas/version_service_client.go

```diff
@@ -208,24 +208,24 @@ func (c *VersionServiceClient) IsOperatorVersionSupported(ctx context.Context, o
 	return false, nil
 }
 
-func latest(m map[string]componentVersion) (*goversion.Version, error) {
+func latestRecommended(m map[string]componentVersion) (*goversion.Version, error) {
 	if len(m) == 0 {
 		return nil, errNoVersionsFound
 	}
 	latest := goversion.Must(goversion.NewVersion("0.0.0"))
-	for version := range m {
+	for version, component := range m {
 		parsedVersion, err := goversion.NewVersion(version)
 		if err != nil {
 			return nil, err
 		}
-		if parsedVersion.GreaterThan(latest) {
+		if parsedVersion.GreaterThan(latest) && component.Status == "recommended" {
```

Comment:
What if there is no recommended?
Shouldn't we return latest recommended or just latest if there is no recommended?

Reply:
at this moment it's hardly possible to not have recommended one. Based on what I heard from the version service team we always should have one recommended version.